### PR TITLE
[QA-329] Adding pipe method.

### PR DESCRIPTION
**Pull request by Aleksandras-V**

### CODE REVIEW

**Review by RomainChaumontet — APPROVED** [review]
Great work!

File: ipc_sendfile.h [hunk 1]
```diff
@@ -3,17 +3,20 @@
...
 
 
 #define IOV_MSG_TYPE (_IO_MAX + 1)
 #define GET_SHMEM_MSG_TYPE (_IO_MAX + 2)
 #define CHANGED_SHMEM_MSG_TYPE (_IO_MAX+3)
 //#define RELEASE_SHMEM_MSG_TYPE (_IO_MAX+4)
+#define PIPE_FD_SEND (_IO_MAX+5)
 #define iov_block_size 4096
 #define semaphore_ready (_PULSE_CODE_MINAVAIL)
+#define pipe_size 4096
```
Comment (ivan-estonia) [inline ipc_sendfile.h]: Never seen defines named with small letters. It is better to use all capital letters.

File: ipc_receivefile.c [hunk 2]
```diff
@@ -348,7 +357,66 @@ int ipc_receive_shm(char* fileName){
...
 
+	unsigned char* buffer;
+
+	printf("Initializing file... \n");
+	fileInit(fileName);
+
+	buffer = malloc(pipe_size);
+	if (buffer== NULL)
+	{
+		printf("ERROR: Malloc");
```
Comment (ivan-estonia) [inline ipc_receivefile.c]:
```suggestion
		fprintf(stderr, "ERROR: Malloc");
```

File: ipc_receivefile.c [hunk 3]
```diff
@@ -348,7 +357,66 @@ int ipc_receive_shm(char* fileName){
...
+		if (status == -1 ){
+			sleep(2);
+		}else{
+			fd = open(pipe_name, O_RDONLY);
+			if (fd == -1){
+				perror ("open pipe");
+				exit(EXIT_FAILURE);
+			}
+		}
+	} while (status == -1);
```
Comment (ivan-estonia) [inline ipc_receivefile.c]:
```suggestion
	while (-1 == access (pipe_name, F_OK)) {
	        sleep (2);					
	}
	
	fd = open(pipe_name, O_RDONLY);
	if (fd == -1) {
		perror ("open pipe");
		exit (EXIT_FAILURE);
	}
```

File: ipc_receivefile.c [hunk 4]
```diff
@@ -348,7 +357,66 @@ int ipc_receive_shm(char* fileName){
...
+	do{
+		part_size=read(fd, buffer, pipe_size);
+		if (part_size == -1)
+		{
+			perror("Pipe read");
+			exit(EXIT_FAILURE);
```
Comment (ivan-estonia) [inline ipc_receivefile.c:396]: `close(pipe_name)`

File: ipc_receivefile.c [hunk 5]
```diff
@@ -348,7 +357,66 @@ int ipc_receive_shm(char* fileName){
...
+		}
+		status=writeFile(buffer,fileName,part_size);//writing message to the file
+		if (status != 0){
+			printf("writeFile Error\n");
+			exit(EXIT_FAILURE);
```
Comment (ivan-estonia) [inline ipc_receivefile.c:403]: `close(pipe_name)`

File: ipc_receivefile.c [hunk 6]
```diff
@@ -348,7 +357,66 @@ int ipc_receive_shm(char* fileName){
...
+	unsigned char* buffer;
+
+	printf("Initializing file... \n");
+	fileInit(fileName);
+
+	buffer = malloc(pipe_size);
+	if (buffer== NULL)
+	{
+		printf("ERROR: Malloc");
+		exit(EXIT_FAILURE);
```
Comment (ivan-estonia) [inline ipc_receivefile.c:376]: free(buffer);

File: ipc_receivefile.c (hunk 4 above)
Comment (ivan-estonia) [reply ipc_receivefile.c:396]: `free(buffer);`

File: ipc_receivefile.c (hunk 5 above)
Comment (ivan-estonia) [reply ipc_receivefile.c:403]: `free(buffer);`

**Review by ivan-estonia — CHANGES_REQUESTED** [review]

File: ipc_receivefile.c (hunk 6 above)
Comment (Aleksandras-V) [reply ipc_receivefile.c:376]: Corrected !

File: ipc_receivefile.c (hunk 5 above)
Comment (Aleksandras-V) [reply ipc_receivefile.c:403]: Corrected !

File: ipc_receivefile.c (hunk 4 above)
Comment (Aleksandras-V) [reply ipc_receivefile.c:396]: Corrected !

File: ipc_sendfile.h (hunk 1 above)
Comment (Aleksandras-V) [reply ipc_sendfile.h]: Corrected !

**Comment by baovu-unikie** [comment]
I pulled the latest commit. Here are the errors:

image

**Review by ivan-estonia — APPROVED** [review]

**Pull request merged**